### PR TITLE
Tool to upgrade GKE clusters

### PR DESCRIPTION
**Pull request by fejta**
```console
$ fejta@fejta3:~/src/gh/test-infra$ bazel run //experiment/cluster-upgrader -- --project=k8s-prow --zone=us-central1-f --cluster=prow --ceiling=1.15.0


Fetching server config for us-central1-f
INFO[0000] Running command                               args="[container clusters upgrade --project=k8s-prow --zone=us-central1-f prow --master --cluster-version=1.14]" command=gcloud
Master of cluster [prow] will be upgraded from version 
[1.13.11-gke.14] to version [1.14]. This operation is long-running and
 will block other operations on the cluster (including delete) until 
it has run to completion.

Do you want to continue (Y/n)?  

Upgrading prow...done.                                                                                                                                                                                                           
Updated [https://container.googleapis.com/v1/projects/k8s-prow/zones/us-central1-f/clusters/prow].
INFO[0286] Master at goal                                cluster=prow masterGoal=1.14.8-gke.17 poolGoal=1.14.8-gke.17 project=k8s-prow zone=us-central1-f
INFO[0287] Running command                               args="[container clusters upgrade --project=k8s-prow --zone=us-central1-f prow --node-pool=default-pool --cluster-version=1.14]" command=gcloud
All nodes in node pool [default-pool] of cluster [prow] will be 
upgraded from version [1.13.11-gke.14] to version [1.14]. This 
operation is long-running and will block other operations on the 
cluster (including delete) until it has run to completion.

Do you want to continue (Y/n)?  y
```

### CODE REVIEW

File: experiment/cluster-upgrader/main.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+// output returns the output and prints Stderr to screen.
+func output(command string, args ...string) (string, error) {
+	logrus.WithFields(logrus.Fields{
+		"command": command,
+		"args":    args,
+	}).Debug("Grabbing output")
+	cmd := exec.Command(command, args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
```
[the remark this answers is not in the document]
Comment (fejta) [reply experiment/cluster-upgrader/main.go]: For some weird reason gcloud now needs me to touch my security every day before running requests, so this is actually useful.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply experiment/cluster-upgrader/main.go]: ACK

File: go.mod [hunk 2]
```diff
@@ -29,6 +29,7 @@ require (
 	github.com/aws/aws-sdk-go v1.23.22
 	github.com/bazelbuild/buildtools v0.0.0-20190917191645-69366ca98f89
 	github.com/beorn7/perks v1.0.1 // indirect
+	github.com/blang/semver v3.5.1+incompatible
```
[the remark this answers is not in the document]
Comment (fejta) [reply go.mod]: Other things are already actually using this (note there isn't a go.sum change) so /shrug

**Pull request merged**